### PR TITLE
HPCC-16833 Regression suite mode to fail on row leaks and missing stops

### PR DESCRIPTION
**Pull request by richardkchapman**
Designed for use from regression suite, option to treat row leaks as fatal, to
try to make sure they are not accidentally introduced.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Pleae review

**Comment by ghalliday** [comment]
@AttilaVamos what scripts need to be changed to ensure that this option is used when we run the regression suite?  Should it be a default option within ecl-test?

**Comment by AttilaVamos** [comment]
@ghalliday I talked about it with Richard yesterday. I will create a new JIRA (today) to add this capability to Regression Test Engine. I presume there can be some more parameters to control engines so my plan is to store them into ecl-test.json config file to avoid code modification every time when we want to modify/extend the engines behaviour.

**Comment by richardkchapman** [comment]
@ghalliday I think the commits here fix the existing failures that this mode causes in the Roxie regression suite - so it should be safe to merge now. The new Roxie flag is pretty safe (as no-one will use it outside of the regression suite) but the changes to fix the leaks are potentially more dangerous!

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -16283,14 +16278,11 @@ class CRoxieServerNWayInputActivity : public CRoxieServerNWayInputBaseActivity
                 selectedJunctions.append(junctionArray[nextIndex-1]);
             }
         }
-
         // NB: Whatever pulls this nwayinput activity, starts and stops the selectedInputs and selectedJunctions
+        // But the N-Way input itself is now donw with processing, and can stop itself/dependencies.
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: typo donw->done

**Pull request merged**